stateproof/worker.go

```diff
@@ -42,6 +42,11 @@ type builder struct {
 	Message   stateproofmsg.Message   `codec:"msg"`
 }
 
+// This is a soft limit on how many builders should be kept in memory, the rest shall be fetched from DB.
+// At most times only 1 should builder should be stored (both in memory and on disk), as this feature
+// is mostly used for recoverability purposes - in case the StateProof chain is stalled.
+const numBuildersInMemory = 15
```

Comment:
We assume the most likely reason the StateProof chain will stall is cryptographic - not enough signatures has been collected to reach the weight threshold required to generate the StateProof. 
The reason we prefer to handle more than 2 builders at once is related to the catchup time of the StateProof chain;

Let's say, for example, one heavy participant was disconnected from the network for a few days and didn't send his signatures. The StateProof chain is stalled since his signatures was needed to reach the required weight.
When this participant comes back online and catches up, he will sign and send his signatures in the process.
In order for the network to be able to handle and verify these signatures, the relevant StateProof builders must be cached in memory. The more builders cached in memory - more signatures could be sent and handled simultaneously and thus the rate in which the relays could build and transmit the missing StateProofs would be faster.

If only 2 builders are in memory at once, the catchup rate shall be 2 StateProof transactions per 128 rounds.
Meaning a chain stalled by a week will take over 2 days to catch up. That is the main reasoning behind storing multiple builders in memory, instead of only the builder for the next required StateProof

I should add that when the StateProof chain is advancing properly the number of builders stored in memory will be exactly 1, as the builders are deleted as soon as their respective StateProof transaction is confirmed on the network.